test/Microsoft.AspNetCore.SignalR.Common.Tests/Internal/Protocol/HubMessageHelpers.cs

```diff
@@ -13,8 +13,14 @@ public static HubMessage AddHeaders(IDictionary<string, string> headers, HubInvo
         {
             foreach (var header in headers)
             {
+                if (hubMessage.Headers == null)
```

Comment:
Move above foreach. Place `if (headers.Count > 0)` around logic to skip if not needed

Reply:
It's a test helper, performance isn't a concern 😄